crates/sui-indexer-alt-graphql/src/api/types/available_range.rs

```diff
@@ -0,0 +1,382 @@
+        ("Query", "epochs") => vec!["kv_epoch_starts"],
+        ("Query", "event") => vec!["ev_struct_inst", "ev_emit_mod"],
+        ("Query", "events") => {
+            let mut pipelines = vec!["ev_struct_inst"];
+            for filter in filters {
+                if filter == "module" || filter == "sender" {
+                    pipelines.push("ev_emit_mod");
+                }
+            }
+            pipelines
+        }
+        ("Query", "multiGetCheckpoints") => pipelines("Query", "checkpoint", None),
+        ("Query", "multiGetEpochs") => pipelines("Query", "epoch", None),
+        ("Query", "multiGetObjects") => pipelines("Query", "object", None),
+        ("Query", "multiGetPackages") => pipelines("Query", "package", None),
+        ("Query", "multiGetTransactionEffects") => pipelines("Query", "transactionEffects", None),
+        ("Query", "multiGetTransactions") => pipelines("Query", "transaction", None),
+        ("Query", "multiGetTypes") => pipelines("Query", "type", None),
+        ("Query", "object") => vec!["obj_versions"],
+        ("Query", "objects") => vec!["consistent"],
+        ("Query", "objectVersions") => vec!["obj_versions"],
+        ("Query", "package") => vec!["kv_packages"],
+        ("Query", "packages") => vec!["kv_packages"],
+        ("Query", "packageVersions") => vec!["kv_packages"],
+        ("Query", "protocolConfigs") => vec!["kv_epoch_starts"],
+        ("Query", "simulateTransaction") => vec![],
+        ("Query", "suinsName") => vec!["obj_versions"],
+        ("Query", "transaction") => vec!["kv_transactions"],
+        ("Query", "transactionEffects") => vec!["kv_transactions"],
+        ("Query", "transactions") => {
+            let mut pipelines = vec!["tx_digests"];
+            for filter in filters {
+                if filter == "affectedAddress" || filter == "sentAddress" || filter == "kind" {
+                    pipelines.push("tx_affected_addresses");
+                }
+                if filter == "kind" {
+                    pipelines.push("tx_kinds")
+                }
+                if filter == "function" {
+                    pipelines.push("tx_calls")
+                }
+                if filter == "affectedObjects" {
+                    pipelines.push("tx_affected_objects")
+                }
+            }
+            pipelines
+        }
+        ("Query", "type") => vec!["kv_packages"],
+
+        // Transaction fields
+        ("Transaction", "digest") => vec!["kv_transactions"],
+        ("Transaction", "effects") => vec!["kv_transactions"],
+        ("Transaction", "expiration") => vec!["kv_transactions"],
+        ("Transaction", "gasInput") => vec!["kv_transactions"],
+        ("Transaction", "kind") => vec!["kv_transactions"],
+        ("Transaction", "sender") => vec!["kv_transactions"],
+        ("Transaction", "signatures") => vec!["kv_transactions"],
+        ("Transaction", "transactionBcs") => vec!["kv_transactions"],
```

Comment:
Thinking about a catch all match to reduce repeated code for some type/field mappings
`("Transaction", _) => vec!["kv_transactions"]`, I feel like this should capture most type and field mappings, the fields that depend on stricter retention can be explicitly matched.